modules/core/src/main/scala/org/scalasteward/core/update/GroupMigrations.scala

```diff
@@ -0,0 +1,125 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.scalasteward.core.update
+
+import cats.implicits._
+import org.scalasteward.core.data._
+import org.scalasteward.core.application.Config
+import org.scalasteward.core.update.GroupMigrations.GroupMigration
+import org.scalasteward.core.io.FileAlg
+import org.scalasteward.core.util.{MonadThrowable, Nel}
+import io.circe.{parser, Decoder}
+import io.circe.generic.extras.{semiauto, Configuration}
+
+trait GroupMigrations[F[_]] {
+  def loadMigrations: F[List[GroupMigration]]
+  def findUpdateWithNewerGroupId(dependency: Dependency): F[Option[Update.Single]]
+}
+
+object GroupMigrations {
+  def create[F[_]: MonadThrowable](
+      implicit fileAlg: FileAlg[F],
+      config: Config,
+      streamCompiler: fs2.Stream.Compiler[F, F]
+  ): GroupMigrations[F] =
+    new GroupMigrations[F] {
+      val _ = fileAlg
+      val _ = config
+
+      val loadMigrations: F[List[GroupMigration]] =
+        config.groupMigrations
+          .foldMap(fs2.Stream.emit)
+          .evalMap(fileAlg.readFile)
+          .unNone
+          .evalMap(parser.decode[List[GroupMigration]](_).liftTo[F])
+          .flatMap(fs2.Stream.emits)
+          .compile
+          .toList
+
+      config.groupMigrations.flatTraverse(fileAlg.readFile).map(_.toList)
+
+      def findUpdateWithNewerGroupId(dependency: Dependency): F[Option[Update.Single]] =
+        loadMigrations.map { extraMigrations =>
+          migrateGroupId(dependency, standardGroupMigrations ++ extraMigrations)
+        }
+    }
+
+  def migrateGroupId(
+      dependency: Dependency,
+      migrations: List[GroupMigration]
+  ): Option[Update.Single] = {
+    val artifactNoCross = dependency.artifactId.copy(maybeCrossName = None)
+
+    migrations.view
+      .filter(_.organizationBefore === dependency.groupId)
+      .find(_.artifactId === artifactNoCross)
+      .map { migration =>
+        Update.Single(
+          CrossDependency(dependency),
+          Nel.one(migration.initialVersion),
+          Some(migration.organizationAfter)
+        )
+      }
+  }
+
+  val standardGroupMigrations: List[GroupMigration] = List(
+    GroupMigration(
+      GroupId("com.geirsson") -> GroupId("org.scalameta"),
+      ArtifactId("sbt-scalafmt"),
+      "2.0.0"
+    ),
+    GroupMigration(
+      GroupId("com.github.mpilquist") -> GroupId("org.typelevel"),
+      ArtifactId("simulacrum"),
+      "1.0.0"
+    ),
+    GroupMigration(
+      GroupId("net.ceedubs") -> GroupId("com.iheart"),
+      ArtifactId("ficus"),
+      "1.3.4"
+    ),
+    GroupMigration(
+      GroupId("org.spire-math") -> GroupId("org.typelevel"),
+      ArtifactId("kind-projector"),
+      "0.10.0"
+    )
+  )
+
+  final case class GroupMigration(
```

Comment:
Can we change `GroupMigration` to `GroupIdChange`? Migration sounds like this would be something sophisticated but we're merely replacing a string with another string. :-) I'd also rename `GroupMigrations` to `GroupIdChangeAlg` or `GroupIdChanges` accordingly.